lib/puppet/defaults.rb

```diff
@@ -1926,4 +1926,24 @@ def self.default_diffargs
           generate manifest documentation.",
     }
   )
+
+  define_settings(
+    :main,
+    :rich_data => {
+      :default  => false,
+      :type     => :boolean,
+      :hook    => proc do |value|
+        envs = Puppet.lookup(:environments) { nil }
+        envs.clear_all unless envs.nil?
+      end,
+      :desc     => <<-'EOT'
+        Enables storage of extended data in the catalog by introducing the new Resource
+        property `ext_parameters`. When enabled, values of type Regexp, SemVer, SemVerRange,
+        Timespan, and Timestamp, will no longer be converted to strings. Instead, they will
+        be serialized using Pcore serialization which allowes them to retain their type. It
+        will also be possible to use parameters that store a Binary.
```

Comment:
A resource has parameters and properties - together they are the attributes of the resource (the expressions that set them have names like AttributeOperation). When serialized they end up under "parameters" irrespective of what kind of attribute they are (although some are properties).
In the sentence I wrote I am also deliberately using 'attributes' as a non technical term to distinguish it from the technical Parameter and Property kinds of attributes. I could even imagine saying "resources containing values of the data types..."